From 85257652417110f49003ab1ae318f89e52128a32 Mon Sep 17 00:00:00 2001
From: Andrey Filipenkov <decapitator@ukr.net>
Date: Sun, 4 Oct 2026 08:39:14 +0300
Subject: [PATCH] [iOS] build RelWithDebInfo, package debug symbols for
 AppStoreConnect

(cherry picked from commit 5260e44c169ada5eeda3d9fef0492b2cd7baa714)
---
 tools/buildsteps/defaultenv  |  2 +-
 tools/buildsteps/ios/package | 10 +++++++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/tools/buildsteps/defaultenv b/tools/buildsteps/defaultenv
index b112fad..b5dc0ed 100644
--- a/tools/buildsteps/defaultenv
+++ b/tools/buildsteps/defaultenv
@@ -34,7 +34,7 @@ case $XBMC_PLATFORM_DIR in
   ios)
     DEFAULT_SDK_VERSION=26.2
     DEFAULT_XBMC_DEPENDS_ROOT=$WORKSPACE/tools/depends/xbmc-depends
-    DEFAULT_CONFIGURATION="Debug"
+    DEFAULT_CONFIGURATION="RelWithDebInfo"
     DEFAULT_XCODE_APP="Xcode-26.3.0.app"
     ;;
 
diff --git a/tools/buildsteps/ios/package b/tools/buildsteps/ios/package
index 587c334..c0ff809 100755
--- a/tools/buildsteps/ios/package
+++ b/tools/buildsteps/ios/package
@@ -2,7 +2,8 @@ WORKSPACE=${WORKSPACE:-$( cd $(dirname $0)/../../.. ; pwd -P )}
 XBMC_PLATFORM_DIR=ios
 . $WORKSPACE/tools/buildsteps/defaultenv
 
-cd $WORKSPACE/build/tools/darwin/packaging/darwin_embedded/
+BUILD_DIR="$WORKSPACE/build"
+cd "$BUILD_DIR/tools/darwin/packaging/darwin_embedded"
 
 #rename for upload
 #e.x. kodi-${JENKINS_BUILD_STRING}-ios64.deb
@@ -12,3 +13,10 @@ UPLOAD_FILENAME="kodi-$(getBuildRevDateStr)-ios64"
 mkdir -p $UPLOAD_DIR
 mv *.deb $UPLOAD_DIR/$UPLOAD_FILENAME.deb
 mv *.ipa $UPLOAD_DIR/$UPLOAD_FILENAME.ipa
+
+# Create AppStore debug symbols
+IOS_APPSTORE_SYMBOLS_DIR=symbols
+mkdir "$IOS_APPSTORE_SYMBOLS_DIR"
+xcrun symbols -arch all -symbolsPackageDir "$IOS_APPSTORE_SYMBOLS_DIR" "$BUILD_DIR/build/$Configuration-iphoneos"/*.dSYM
+tar -cf "$UPLOAD_DIR/${UPLOAD_FILENAME}_symbols.txz" --xz "$IOS_APPSTORE_SYMBOLS_DIR"
+rm -rf "$IOS_APPSTORE_SYMBOLS_DIR"
-- 
2.53.0

